app.ts

```diff
@@ -0,0 +1,140 @@
+﻿// Onboarding Project - 2018
+
+const RECORD_COUNT = JSON.parse(getData("http://localhost:2050/recordCount")); // total number of records available in webserver
+let LAST_ENTRY, CURRENT_ENTRY; // entry numbers required for prev and next funtionality
+let resizeTimer; // delay required to reduce the number of calls made by browser when resizing the window
+let CONTROLS_HEIGHT;
+
+// makes api call and returns data
+function getData(dataLink: string) {
+    let apiCall = new XMLHttpRequest();
+    apiCall.open("GET", dataLink, false); // false to prevent asych // warning need to change
+    apiCall.send(null);
+    return apiCall.responseText;
+}
+
+// converts strings into array and table
+function parseText(textString) {
+    let outputArray = textString.replace(/'/g, '""');
+    outputArray = JSON.parse(textString);
```

Comment:
This is naughty, I'm going to leave it to you to figure out why ^^

Reply:
typo